### PR TITLE
BL-578_api-error-handling

### PR DESCRIPTION
**Pull request by jack-semder**
## Description

- Added specific error message for when entries are created with duplicate profile IDs.
- Removed Internal Service Error for create/mentee and create/mentor endpoints

## Type of change

- [✅] New feature

## Checklist:

- [✅] My code follows PEP8 style guide
- [✅] I have removed unnecessary print statements from my code
- [✅] I have made corresponding changes to the documentation if necessary
- [✅] My changes generate no errors
- [✅] No commented-out code
- [✅] Size of pull request kept to a minimum
- [✅] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Comment by jack-semder** [comment]
Here's a Loom video!

https://www.loom.com/share/ab807cfa61a944d9a1e23cfaaa00e6b1

File: app/api.py [hunk 1]
```diff
@@ -109,7 +113,10 @@ async def create_mentee(data: Mentee):
     Returns:
         New record data or schema discrepancy error as dictionary
     """
-    return {"result": API.db.create("Mentees", data.dict())}
+    try:
+        return {"result": API.db.create("Mentees", data.dict())} 
+    except DuplicateKeyError:
+        raise HTTPException(status_code=409, detail="Profile ID must be unique.")
```
Comment (woodbenwood) [inline app/api.py:120]: Elegant, lean coding. An improvement from the prior version.

Comment (jack-semder) [reply app/api.py:120]: Hey thanks Ben

**Review by demetriustuck — APPROVED** [review]
Sleek! Good improvements from last time. Precise error code.

**Review by woodbenwood — APPROVED** [review]

**Review by CrisIV — APPROVED** [review]
Not knowing the reason why something breaks when it does is the worst! Providing clarity on that can save a lot of time and frustration, great job!

**Review by L0lbrich — APPROVED** [review]
very important features, glad to see were moving in this direction

**Pull request merged**